beard/similarity/pairs.py

```diff
@@ -464,3 +465,125 @@ def transform(self, X):
             Xt[np.where(denominator == 0)[0]] = 0.
 
         return np.array(Xt).reshape(-1, 1)
+
+
+class CharacterEquality(BaseEstimator, TransformerMixin):
+
+    """Equality of characters on paired data.
+
+    Returns 1 in case of equality, 0 otherwise.
+
+    In case empty characters are provided, a uncertanity value (0.5) is output.
+    """
+
+    def fit(self, X, y=None):
+        """(Do nothing).
+
+        Parameters
+        ----------
+        :param X: array-like, shape (n_samples, n_features)
+            Input data.
+
+        Returns
+        -------
+        :returns: self
+        """
+        return self
+
+    def transform(self, X):
+        """Compute equality.
+
+        Rows i in ``X`` are assumed to represent pairs, where
+        ``X[i, :n_features]`` and ``X[i, n_features:]`` correspond to their two
+        individual elements, each representing a character.
+
+        Parameters
+        ----------
+        :param X: array-like, shape (n_samples, n_features)
+            Input data.
+
+        Returns
+        -------
+        :returns: Xt array-like, shape (n_samples, 1)
+            The transformed data.
+        """
+        def character_equality(x, y):
+            if x != y:
+                return 0.
+            elif x == "":
+                return 0.5
+            else:
+                return 1.
+
+        X1, X2 = np.split(X, 2)
+
+        vectorized = np.vectorize(character_equality)
+        n_samples = X1.shape[0]
+
+        val = vectorized(X1, X2)
+        return val.reshape((n_samples, 1))
+
+
+class StringDistance(BaseEstimator, TransformerMixin):
```

Comment:
This class be used to obtain the same as `CharacterEquality` above, by providing the proper similarity function. I would therefore simply remove `CharacterEquality`.